deepdiff/helper.py

```diff
@@ -471,6 +472,27 @@ def literal_eval_extended(item):
         raise
 
 
+def time_to_seconds(t):
+    return (t.hour * 60 + t.minute) * 60 + t.second
+
+
+def datetime_normalize(truncate_datetime, obj):
+    if truncate_datetime:
+        if truncate_datetime == 'second':
+            obj = obj.replace(microsecond=0)
+        elif truncate_datetime == 'minute':
+            obj = obj.replace(second=0, microsecond=0)
+        elif truncate_datetime == 'hour':
+            obj = obj.replace(minute=0, second=0, microsecond=0)
+        elif truncate_datetime == 'day':
+            obj = obj.replace(hour=0, minute=0, second=0, microsecond=0)
+    if isinstance(obj, datetime.datetime):
+        obj = obj.replace(tzinfo=datetime.timezone.utc).timestamp()
```

Comment:
Hi @timson 
I know I have already merged the PR. While I was playing around with this new feature, a question came up for me. Why do you convert it to timestamp? If we keep it as a datetime object, it will be more aligned with what I think the users will expect, no? The original objects were datetimes but then in the report we are showing floats.

Reply:
Yes, you right, it is not necessary to convert datetime objects to timestamp, I did it because I thought that I need integer or string object for hashing with DeepHash, but I was wrong )

Reply:
Ok cool. If you don't mind, I'm gonna remove the timestamp conversion then!

Reply:
Also I think the timestamp comparison later might need to be revisited for timezones. For example if 2 datetimes have different timezones. I'm not sure if the current code will produce the correct results. We manually convert the time to be UTC but it might have been something else.

Reply:
I think datetime_normalize function could be rewrited in this way:

```python
def datetime_normalize(truncate_datetime, obj):
    if truncate_datetime:
        if truncate_datetime == 'second':
            obj = obj.replace(microsecond=0)
        elif truncate_datetime == 'minute':
            obj = obj.replace(second=0, microsecond=0)
        elif truncate_datetime == 'hour':
            obj = obj.replace(minute=0, second=0, microsecond=0)
        elif truncate_datetime == 'day':
            obj = obj.replace(hour=0, minute=0, second=0, microsecond=0)
    return obj
```
Just truncation without conversion to utc timestamps. Also we don't need convert time object to seconds anymore.